testsuite/replay_verify.py

```diff
@@ -71,6 +71,25 @@
 ]
 
 
+# retry the replay_verify_partition if it fails
+def retry_replay_verify_partition(func, *args, **kwargs) -> Tuple[int, int, bytes]:
+    res = (0, 0, b"")
+    NUM_OF_RETRIES = 3
+    i = 1
+    while i < NUM_OF_RETRIES:
+        i += 1
+        (partition_number, code, msg) = func(*args, **kwargs)
+        print(f"[partition {partition_number}] trying {i}th time")
```

Comment:
first print will be "2th"? 😂 

And I mean.. res doesn't need to be kept across iterations if we add the return under the `code == 1` branch. (updated my previous comment to use `while True:`, which makes more sense.

Reply:
there is some poetry code inspection in CI/CD and enforce there is a return from this function, since it doesn't know if code == 1 would eventually return.

Reply:
okay.. then
1. `print("try {i}")` should be before `func(...)`
2. looks nicer to do:
```python
(partition_number, code, msg) = (None, None, None)

for i in range(1, NUM_TRIES+1):
    print("try {i}")
    (partition_number, code, msg) = func(*args, **kwargs)
    if code != 1:
        break

return (partition_number, code, msg)
```

Reply:
looks beautify